### DISCUSSION TITLE
Custom resource names for cloud provider(Azure)

### DISCUSSION THREAD

**Discussion by sbalajisivaram** [nitrictech/nitric#424] [Q&A] [answered]
Team,
I recently bumped into Nitric documentation as i found this one interesting. As i just started i would like to have this question answered. 1) Do we have an option to provide custom names for the resources we create? I noticed nitirc provisions resource group and other resources on its own but do we get the control to choose our own names for the resource? I know there is a deployment config file for cloud provisioning but not sure how can i utilize that. Any help would be much appreciated.

**Comment by jyecusch** [comment, marked as answer]
Hi @sbalajisivaram

Application level resources are all named by the developer, so that they're easy to reference and recognise. For example, in this code snippet I'm creating an API and a Bucket, giving them both names.

```javascript
import { api, bucket } from "@nitric/sdk";

// create a new API named 'photos'
const photoApi = api('photos');
// create a new Bucket named 'photo-files'
const imgs = bucket('photo-files').for('reading');
```

During deployment these names may be modified to ensure they don't collide with existing resources in your cloud account. Nitric takes care of the name resolution at runtime, so you don't need to be conserned about those renames.

There are other resources deployed by Nitric that are needed to facilitate applications which aren't explicitly defined by the developer (such as Roles, Containers, etc.), those resources can't be named currently. Instead, they get a name derived from context, for example they will be named after your project or the funtion they're associated with.

I'm curious if you need another naming feature I haven't mentioned and what you're trying to achieve. We may be able to help further.

**Comment by sbalajisivaram** [comment]
Thanks @jyecusch , This is helpful and understanding. One other thing, I noticed with cloud provider ```stack``` file we can define configs for the respective resources the one i noticed was compute memory. Likewise, do we have this facilities for all the application level resources. For example, ```collections```?

**Reply by jyecusch** [reply]
Not all resources have that level of configuration available yet, but we're in the process of adding more configuration options at the stack level. If there is a specific configuration you would like for `collections` feel free to open a [feature request](https://github.com/nitrictech/nitric/issues/new?assignees=&labels=enhancement&template=2.feature.md) or submit a PR, we'd be happy to review either :)
